File: python/gtk4/procedural/application-window.py
import gi, sys
gi.require_version("Gtk", "4.0")
from gi.repository import Gtk

APP_ID = "io.github.Miqueas.GTK-Examples.Python.Gtk4.Procedural.ApplicationWindow"
APP_TITLE = "Gtk.ApplicationWindow"

def onAppStartup(self):
    global APP_TITLE

    window = Gtk.ApplicationWindow(application = self)
    window.set_title(APP_TITLE)
    window.set_default_size(400, 400)

if __name__ == "__main__":
    app = Gtk.Application(application_id = APP_ID, flags = 0)
    app.connect("activate", lambda self: self.get_active_window().present())
    app.connect("startup", onAppStartup)

    sys.exit(app.run(sys.argv))

File: cpp/gtk4/procedural/application-window.cpp
#include <gtkmm.h>

static void on_app_activate();
static void on_app_startup();

static const Glib::ustring APP_ID = "io.github.Miqueas.GTK-Examples.C.Gtk4.Procedural.ApplicationWindow";
static const Glib::ustring APP_TITLE = "Gtk::ApplicationWindow";
static Glib::RefPtr<Gtk::Application> app;

int main(int argc, char** argv) {
  app = Gtk::Application::create(APP_ID);
  app->signal_startup().connect(sigc::ptr_fun(&on_app_startup));
  app->signal_activate().connect(sigc::ptr_fun(&on_app_activate));
  return app->run(argc, argv);
}

static void on_app_activate() {
  Gtk::Window* window = app->get_active_window();
  if (window != nullptr) window->present();
}

static void on_app_startup() {
  auto window = Gtk::make_managed<Gtk::ApplicationWindow>(app);
  window->set_title(APP_TITLE);
  window->set_default_size(400, 400);
}